src/vision/oak/transform_data.py

```diff
@@ -26,43 +26,64 @@ def make_sectors(array_size: int, sector_size: float) -> np.ndarray:
     combined_matrix[..., 0, 1] = xv + sector_size
     combined_matrix[..., 1, 0] = yv - sector_size
     combined_matrix[..., 1, 1] = yv
+    path = os.path.join(path, 'sectors.npy')
+    np.save(path, combined_matrix)
 
-    return combined_matrix
 
-
-def assignment_to_sectors(pcd: o3d.geometry.PointCloud):
+def assignment_to_sectors(pcd: o3d.geometry.PointCloud, sector_size: float = 0.01, path: str = './src/vision/oak/sectors.npy'):
     """
+    Assigns points from a point cloud to sectors and calculates the center of mass for each sector.
+
+    The function takes a 3D point cloud, divides the space into sectors of a fixed size, and computes the
+    center of mass for the points in each sector. The result is a 2D matrix where each cell represents
+    a sector and contains the center of mass value for that sector.
+
+    Parameters:
+    pcd (o3d.geometry.PointCloud): The input point cloud containing 3D points.
+    path (str): File path where the sectors numpy array is loaded from. Default is './src/vision/oak/sectors.npy'.
 
-    :param pcd:
-    :return:
+    Returns (np.ndarray): A 2D numpy array where each cell contains the center of mass value for the corresponding sector.
+                The array size is determined by the maximum sector indices found in the point cloud.
     """
-    sector_size = 0.01
-    sectors = np.load('./vision/oak/data.npy')
+    sectors = np.load(path)
     points = np.asarray(pcd.points)
 
-    # indeksy x, z dla każdego punktu
+    # Calculate x and z indices for each point
```

Comment:
Mam taką rozkminę  czy jeśli okaże się ze wartości na których musimy operować są od 0 do 500 bo tak np wychodzą z kamerki i imu ale ustawimy sobie wtedy sector size na 0.1 to będzie tak samo działać i powstanie nam też macierz 5k x 5k ?